extensions/amp-sidebar/1.0/component.js

```diff
@@ -135,42 +137,88 @@ function SidebarWithRef(
   }, [opened, close]);
 
   return (
-    mounted && (
-      <>
-        <ContainWrapper
-          as={Comp}
-          ref={sidebarRef}
-          size={false}
-          layout={true}
-          paint={true}
-          part="sidebar"
-          wrapperClassName={`${classes.sidebar} ${
-            classes.defaultSidebarStyles
-          } ${side === Side.LEFT ? classes.left : classes.right}`}
-          role="menu"
-          tabindex="-1"
-          hidden={!side}
-          {...rest}
-        >
-          {children}
-        </ContainWrapper>
-        <div
-          ref={backdropRef}
-          onClick={() => close()}
-          part="backdrop"
-          style={backdropStyle}
-          className={`${backdropClassName ?? ''} ${classes.backdrop} ${
-            classes.defaultBackdropStyles
-          }`}
-          hidden={!side}
-        >
-          <div className={classes.backdropOverscrollBlocker}></div>
-        </div>
-      </>
-    )
+    <div className={objstr({[classes.unmounted]: !mounted})} part="wrapper">
+      <ContainWrapper
+        as={Comp}
+        ref={sidebarRef}
+        size={false}
+        layout={true}
+        paint={true}
+        part="sidebar"
+        wrapperClassName={objstr({
+          [classes.sidebar]: true,
+          [classes.defaultSidebarStyles]: true,
+          [classes.left]: side === Side.LEFT,
+          [classes.right]: side !== Side.LEFT,
+        })}
+        role="menu"
+        tabindex="-1"
+        hidden={!side}
+        {...rest}
+      >
+        {children}
+      </ContainWrapper>
+      <div
+        ref={backdropRef}
+        onClick={() => close()}
+        part="backdrop"
+        style={backdropStyle}
+        className={objstr({
+          [classes.backdrop]: true,
+          [classes.defaultBackdropStyles]: true,
+          [backdropClassName]: backdropClassName,
+        })}
+        hidden={!side}
+      >
+        <div className={classes.backdropOverscrollBlocker}></div>
+      </div>
+    </div>
   );
 }
 
 const Sidebar = forwardRef(SidebarWithRef);
 Sidebar.displayName = 'Sidebar'; // Make findable for tests.
 export {Sidebar};
+
+/**
+ * @param {!SidebarDef.SidebarToolbarProps} props
+ * @return {PreactDef.Renderable}
+ */
+export function SidebarToolbar({toolbar, toolbarTarget, children, ...rest}) {
+  const ref = useRef();
+  const [matches, setMatches] = useState(false);
+  const [targetEl, setTargetEl] = useState(null);
+
+  useEffect(() => {
+    const window = ref.current?.ownerDocument?.defaultView;
+    if (!window) {
+      return;
+    }
+
+    const mediaQueryList = window.matchMedia(toolbar);
```

Comment:
Can we rename `toolbar` to be more descriptive? It's a media query, right?

Reply:
That works, I've updated it to `mediaQueryProp` internally (within the component).  I think we should keep the prop name `toolbar` externally since it matches with the existing API for `0.1`.  (In 0.1 the user would specify a toolbar as such:) 

```
<nav toolbar="(media-query)" toolbar-target="elementID">
```